tests/php/src/Cli/ReferenceSiteImporter.php

```diff
@@ -0,0 +1,237 @@
+
+		if ( ! $file_name ) {
+			$file_name = md5( $url );
+		}
+
+		$tmp_file_name = wp_tempnam( $file_name );
+		if ( ! $tmp_file_name ) {
+			return new WP_Error( 'import_no_file', __( 'Could not create temporary file.', 'amp' ) );
+		}
+
+		$context_options = [
+			'ssl' => [
+				'verify_peer'      => false,
+				'verify_peer_name' => false,
+			],
+		];
+
+		file_put_contents( // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents -- Needed for stream wrapper support.
+			$tmp_file_name,
+			file_get_contents( // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- Needed for stream wrapper support.
+				$url,
+				false,
+				stream_context_create( $context_options )
+			)
+		);
+
+		// Handle the upload like _wp_handle_upload() does.
+		$wp_filetype     = wp_check_filetype_and_ext( $tmp_file_name, $file_name );
+		$ext             = empty( $wp_filetype['ext'] ) ? '' : $wp_filetype['ext'];
+		$type            = empty( $wp_filetype['type'] ) ? '' : $wp_filetype['type'];
+		$proper_filename = empty( $wp_filetype['proper_filename'] ) ? '' : $wp_filetype['proper_filename'];
+
+		// Check to see if wp_check_filetype_and_ext() determined the filename was incorrect.
+		if ( $proper_filename ) {
+			$file_name = $proper_filename;
+		}
+
+		if ( ( ! $type || ! $ext ) && ! current_user_can( 'unfiltered_upload' ) ) {
+			return new WP_Error( 'import_file_error', __( 'Sorry, this file type is not permitted for security reasons.', 'amp' ) );
+		}
+
+		$uploads = wp_upload_dir( $post['upload_date'] );
+		if ( ! ( $uploads && false === $uploads['error'] ) ) {
+			return new WP_Error( 'upload_dir_error', $uploads['error'] );
+		}
+
+		// Move the file to the uploads dir.
+		$file_name     = wp_unique_filename( $uploads['path'], $file_name );
+		$new_file      = $uploads['path'] . "/$file_name";
+		$move_new_file = copy( $tmp_file_name, $new_file );
+
+		if ( ! $move_new_file ) {
+			@unlink( $tmp_file_name ); // phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged -- Quietly clean up left-overs.
+			return new WP_Error( 'import_file_error', __( 'The uploaded file could not be moved', 'amp' ) );
+		}
+
+		// Set correct file permissions.
+		$stat  = stat( dirname( $new_file ) );
+		$perms = $stat['mode'] & 0000666;
+		chmod( $new_file, $perms );
+
+		$upload = [
+			'file'  => $new_file,
+			'url'   => "{$uploads['url']}/{$file_name}",
+			'type'  => $wp_filetype['type'],
+			'error' => false,
+		];
+
+		// Keep track of the old and new urls so we can substitute them later.
+		$this->url_remap[ $url ]          = $upload['url'];
+		$this->url_remap[ $post['guid'] ] = $upload['url'];
+
+		return $upload;
+	}
+
+	/**
+	 * Check whether the provided URL is pointing to Google Storage.
+	 *
+	 * @param string $url URL to check.
+	 * @return bool Whether this is a Google Storage URL.
+	 */
+	private function is_google_storage_url( $url ) {
+		return 1 === preg_match( '#^gs://#', $url );
```

Comment:
`preg_match` seems like a bit of overkill for this. Something like `strncmp` should do.like in: https://github.com/ampproject/amp-wp/blob/1f1ebeef03862e7a72716f182dc761e9690e732e/tests/php/src/Cli/Export/ExportWxrFile.php#L81

Reply:
https://github.com/ampproject/amp-wp/pull/5162/commits/31d773146f249e2a779b8c22167f4a9acc4752e6